packages/contracts/src/CredbullFixedYieldVaultWithUpside.sol

```diff
@@ -3,9 +3,10 @@ pragma solidity ^0.8.19;
 
 import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
 import { UpsideVault } from "./vaults/UpsideVault.sol";
+import { ICredbull } from "./interface/ICredbull.sol";
 
 contract CredbullFixedYieldVaultWithUpside is UpsideVault {
-    constructor(VaultParams memory params, IERC20 _token, uint256 _collateralPercentage)
+    constructor(ICredbull.FixedYieldVaultParams memory params, IERC20 _token, uint256 _collateralPercentage)
```

Comment:
suggest to group the token and collateral percentage here into a UpsideVaultParams struct.

1. suggest to rename token to something more explicit.  tokenForLock or similar.  having it within the UpsideVaultParams struct will also help with context.  we have so many tokens on Vaults, it isn't obvious what this is!

1. what is the collateralPercentage?  is this the upsideReward that we get?  or tokensRequiredForLock?  let's rename it as appropriate.